src/modules/item/sagas.ts

```diff
@@ -245,8 +283,37 @@ export function* itemSaga(legacyBuilder: LegacyBuilderAPI, builder: BuilderClien
     }
   }
 
-  function* handleSaveItemSuccess() {
-    yield put(closeModal('EditItemURNModal'))
+  function* fetchNewCollectionItemsPaginated(collectionId: string, newItemsAmount = 1) {
+    const paginationData: ItemPaginationData = yield select(getPaginationData, collectionId)
+    const { currentPage, limit, total } = paginationData
+    const newItemPage = Math.ceil((total + newItemsAmount) / limit) // optimistic computation, in case the save is successfull
+    if (newItemPage !== currentPage) {
+      yield put(push(locations.thirdPartyCollectionDetail(collectionId, { page: newItemPage })))
+    } else {
+      yield put(fetchCollectionItemsRequest(collectionId, { page: currentPage, limit }))
+    }
+  }
+
+  function* handleSaveMultipleItemsSuccess(action: SaveMultipleItemsSuccessAction) {
+    const { items } = action.payload
+    const collectionId = items[0].collectionId!
+    const location: ReturnType<typeof getLocation> = yield select(getLocation)
+    if (location.pathname === locations.thirdPartyCollectionDetail(collectionId)) {
+      yield call(fetchNewCollectionItemsPaginated, collectionId, items.length)
+    }
+  }
+
+  function* handleSaveItemSuccess(action: SaveItemSuccessAction) {
+    const openModals: ModalState = yield select(getOpenModals)
+    if (openModals['EditItemURNModal']) {
+      yield put(closeModal('EditItemURNModal'))
+    }
+    const { item } = action.payload
+    const collectionId = item.collectionId!
+    const location: ReturnType<typeof getLocation> = yield select(getLocation)
+    if (location.pathname === locations.thirdPartyCollectionDetail(collectionId)) {
+      yield call(fetchNewCollectionItemsPaginated, collectionId)
+    }
```

Comment:
Sure
Let's say the user is looking at the 5th page (of a total of 20 pages) in the TP collection detail page and the user proceeds to create add a new item. We don't know where that item is going to be.. we can't tell if it's going to be in page 6 or maybe even further in the pages. In order to avoid maintaining the pagination in the UI, I added an extra fetch so the backend replies with the new pagination data.